cookbooks/bcpc-hadoop/recipes/hadoop_config.rb

```diff
@@ -12,6 +12,18 @@
     update-alternatives --set hadoop-conf /etc/hadoop/conf.#{node.chef_environment}
   }
 end
+if ( node[:bcpc][:hadoop][:hdfs][:ldap][:integration] == true )
+
+  ldap_pwd = ( node[:bcpc][:hadoop][:hdfs][:ldap][:password].nil? ? get_config('password', 'ldap', 'os') : node[:bcpc][:hadoop][:hdfs][:ldap][:password] )
+
+  file "/etc/hadoop/conf/ldap-conn-pass.txt" do
+    content "#{ldap_pwd}"
+    mode 0666
```

Comment:
The configuration of `LDAP` is required on each node to make it work properly. For `HDFS` it gets information from `NameNode` but for `HBase` shell `core-site.xml` is processed locally and if `LDAP` configuration is not present it won't honor `LDAP` groups. Since `HBASE` shell is launched by regular user the password file must be world readable.